spec/enumerables_spec.rb

```diff
@@ -1,25 +1,255 @@
 require './enumerables.rb'
 
 describe 'Enumerable' do
-
-  let(:num_array) { [1,3,4,5,6,7,8,5,5] }
+  let(:num_array) { [2, 3, 4, 5, 6, 7, 8, 5, 5] }
   let(:empty_array) { [] }
-  let(:word_array) { ["Sharon", "Leo", "Leila", "Brian", "Arun"] }
-
+  let(:word_array) { %w[Sharon Lea Leila Brian Arun] }
+  let(:name_hash) { { name: 'kedir', last: 'Abdu' } }
   describe '#my_each' do
-    
     context 'If block is not given' do
       it 'returns an enumerator' do
         expect(num_array.my_each.is_a?(Enumerable)).not_to be false
       end
     end
 
     context 'If a block is given' do
-      it 'iterates squares each number in the array' do
+      it 'it squares each number in the array' do
         custom_array = []
         test_array = []
-        num_array.my_each { |i| custom_array << i ** 2 }
-        num_array.each { |i| test_array << i ** 2 }
+        num_array.my_each { |i| custom_array << i**2 }
+        num_array.each { |i| test_array << i**2 }
+        expect(test_array).to eql(custom_array)
+      end
+
+      it 'it prints the contents of the hash' do
+        custom_hash = []
+        test_hash = []
+
+        name_hash.my_each { |k, v| custom_hash << [k.to_s, v.upcase] }
+        name_hash.each { |k, v| test_hash << [k.to_s, v.upcase] }
+        expect(test_hash).to eql(custom_hash)
+      end
+    end
+  end
+
+  describe '#my_each_with_index' do
+    let(:arr) { %w[cat dog wombat] }
+
+    it 'should behave as each_with_index' do
+      hash_b = {}
+      hash_a = {}
+      arr.each_with_index do |item, index|
+        hash_b[item] = index
+      end
+      arr.my_each_with_index do |item, index|
+        hash_a[item] = index
+      end
+      expect(hash_b).to eql(hash_a)
+    end
+  end
+
+  describe '#my_map' do
+    # context 'If block is not given' do
+    #   expect(word_array.my_map.is_a?(Enumerable)).not_to be false
+    # end
```

Comment:
Please remove commented out code. 
As can be seen from [this screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)
It is better to delete commented code. If you need to review a previous version of your code, use git history. 🙂